guide/voice/understanding-voice.md

```diff
@@ -0,0 +1,55 @@
+- [Opus](https://en.wikipedia.org/wiki/Opus_(audio_format)) - This is a _lossy audio format_, it's an encoding applied to PCM that makes music playable over Discord. An Opus encoder generates Opus packets which can be played over Discord.
+- [Ogg](https://en.wikipedia.org/wiki/Ogg) and [WebM](https://en.wikipedia.org/wiki/WebM) - these are audio containers. Think of this as ZIP vs RAR - they both contain the exact same audio, but they go about containing it in different ways. In this context, these containers store lots and lots of Opus audio packets, which can be extracted to be played.
+- [FFmpeg](https://ffmpeg.org/) - for cases where we _don't_ have Opus audio, e.g. in MP3 files, we need a way of converting them to Opus. FFmpeg is the tool we use, it converts hundreds of formats of media to Opus for us so they can be played over Discord.
+
+### The voice connection lifecycle
+1. A packet is sent to the main gateway telling Discord we wish to join a voice channel. Discord responds with a voice gateway (WebSocket) to connect to and the bot authenticates itself. We also select an _encryption mode_ available on that gateway. We now start heartbeating.
+2. We find our external port and send it to Discord over UDP so we can send/receive audio. It follows that a voice connection has two main components, a WebSocket connection - where we can change channels and speaking status - and a UDP socket, where audio is actually sent and received.
+3. We send our audio packets at regular time intervals to the UDP socket, which forwards it to other users connected to the channel.
+4. The server may change its voice region, in this case Discord.js will disconnect these sockets and reconnect to the new voice servers.
+5. Once we're done with the voice connection, we tell the main gateway that we're no longer in a voice channel, and we can now destroy the UDP socket and voice gateway connection without disrupting any activity on the normal gateway.
+
+### Playing audio
+
+To play audio, we need to send Opus audio packets to Discord at a regular time interval - we have selected 20ms. This is the job of the `StreamDispatcher` - it is simply a WritableStream that has Opus audio packets written to it. The dispatcher handles the timing of the packets, and also applies some metadata, e.g. the sequence number of the packet, the timestamp, and then encrypts the packet before sending it via the UDP socket.
+
+Obtaining the Opus audio is mainly offloaded to a separate module, [`prism-media`](https://github.com/amishshah/prism-media). This module converts media to Opus audio, either through "demuxing" Ogg/WebM media files (extracting Opus audio directly), or using FFmpeg to transcode most media formats to Opus audio so we can use it.
+
+It also can be used to change the volume of a stream in real-time without much cost to performance.
```

Comment:
"It can also be used" sounds more fluid than "It also can be used" in my opinion.